apm-sniffer/apm-sdk-plugin/rabbitmq-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/rabbitmq/define/RabbitMQConsumerInstrumentation.java

```diff
@@ -71,6 +71,6 @@
 
     @Override
     protected ClassMatch enhanceClass() {
-        return MultiClassNameMatch.byMultiClassMatch(ENHANCE_CLASS_PRODUCER);
+        return HierarchyMatch.byHierarchyMatch(new String[] {ENHANCE_CLASS_PRODUCER}, true);
```

Comment:
`byHierarchyMatch` doesn't recommend, too. If have to do, I think you should explain and discuss more. :)

Reply:
Yes. We don't recommend because of potential side effect. Such as someone implement the interfaces.

Reply:
See my note above that explains why we should do this